packages/beacon-node/src/chain/bls/multithread/jobItem.ts

```diff
@@ -70,6 +73,42 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
       // and not a problem in the near future
       // this is monitored on v1.11.0 https://github.com/ChainSafe/lodestar/pull/5912#issuecomment-1700320307
       const timer = metrics?.blsThreadPool.signatureDeserializationMainThreadDuration.startTimer();
+
+      // adding verification randomness is swig specific. must not attempt with herumi until
+      // @chainsafe/bls is updated to support it with herumi
+      if (job.opts.addVerificationRandomness) {
+        const pkPoint = new swigBindings.blst.P1();
+        const sigPoint = new swigBindings.blst.P2();
+        for (let i = 0; i < job.sets.length; i++) {
```

Comment:
Looks like we can iterate over set directly, bypassing indexes?

Reply:
good call!!! copy/pasta. Updated!